queryable-spec.ts

```diff
@@ -0,0 +1,390 @@
+ * operator on given sub-expressions.
+ * 
+ * The WebIDL definition of the Filterable spec contains a list of supported 
+ * operators: https://rdf.js.org/query-spec/#expression-operators 
+ */
+interface OperatorExpression extends Expression {
+  
+  /** 
+   * Contains the constant "operator". 
+   */
+  expressionType: 'operator';
+
+  /**
+   * Value that identifies an operator. Possible values can be found in the 
+   * list of operators.
+   */
+  operator: string;
+
+  /**
+   * Array of Expression's on to which the given operator applies. The length
+   * of this array depends on the operator.
+   */
+  args: Expression[];
+};
+
+/**
+ * A TermExpression is an expression that contains a Term.
+ */
+interface TermExpression {
+  
+  /**
+   * The constant "term".
+   */
+  expressionType: 'term';
+
+  /**
+   * a Term.
+   */
+  term: RDF.Term;
+}
+
+/**
+ * ExpressionFactory enables expressions to be created in an idiomatic manner.
+ */
+interface ExpressionFactory {
+  
+  /**
+   * Creates a new OperatorExpression instance for the given operator and array of arguments.
+   */
+  operatorExpression(operator: string, args: Expression[]): OperatorExpression;
+
+  /**
+   * Creates a new TermExpression instance for the given term.
+   */
+  termExpression(term: RDF.Term): TermExpression;
+};
+
+
+/**
+ * QueryResultMetadataCount is part of the QueryResultMetadata interface to
+ * represent metadata about the number of quads in the result stream.
+ */
+interface FilterableResultMetadataCount {
+  /**
+   * indicates the type of counting that was done, and MUST either be 
+   * "estimate" or "exact".
+   */
+  type: 'estimate' | 'exact';
+
+  /**
+   * Indicates an estimate of the number of quads in the stream if 
+   * type = "estimate", or the exact number of quads in the stream if 
+   * type = "exact".
+   */
+  value: number;
+};
+
+/**
+ * TBD
+ */
+ interface FilterableCost {
+  /**
+   * An estimation of how many iterations over items are executed.
+   * This is used to determine the CPU cost.
+   */
+  iterations: number;
+  /**
+   * An estimation of how many items are stored in memory.
+   * This is used to determine the memory cost.
+   */
+  persistedItems: number;
+  /**
+   * An estimation of how many items block the stream.
+   * This is used to determine the time the stream is not progressing anymore.
+   */
+  blockingItems: number;
+  /**
+   * An estimation of the time to request items from sources.
+   * This estimation can be based on the `cardinality`, `pageSize`, and `requestTime` metadata entries.
+   * This is used to determine the I/O cost.
+   */
+  requestTime: number;
+}
+
+/**
+ * TBD
+ */
+interface FilterableOrder {
+  cost: FilterableCost;
+  terms: { term: TermName, direction: 'asc' | 'desc' }[];
+}
+  
+/**
+ * A QueryResultMetadata is an object that contains metadata about a certain
+ * query result, such as invoking FilterableSource.matchExpression.
+ */
+interface FilterableResultMetadata {
```

Comment:
```suggestion
interface FilterableResultMetadata extends Record<string, any> {
```
Also for extensibility?

Reply:
Were we going to add `[key: string]: any;` here instead?